ouroboros-network-framework/test/Test/Simulation/Network/Snocket.hs

```diff
@@ -627,28 +638,208 @@ prop_generator_NonFailingBeararInfoScript (NonFailingBearerInfoScript s) = not (
 
 prop_client_server :: [ByteString] -> BearerInfoScript -> Property
 prop_client_server payloads (BearerInfoScript script) =
-    let tr = runSimTrace $ clientServerSimulation script' payloads
-    in -- Debug.traceShow script $
-       case traceResult True tr of
-         Left e         -> counterexample
-                             (unlines
-                               [ "=== Say Events ==="
-                               , unlines (selectTraceEventsSay' tr)
-                               , "=== Error ==="
-                               , show e ++ "\n"
-                               , "=== Trace Events ==="
-                               , unlines (show `map` traceEvents tr)
-                               ])
-                             False
-         Right Nothing  -> property False
-         Right (Just b) -> property b
+  prop_verify_connect_to_simulation sim
   where
+    sim :: forall s . IOSim s (Either SomeException ())
+    sim = clientServerSimulation script' payloads
+
     script' = toBearerInfo <$> script
 
+data TestError = UnexpectedOutcome
+  deriving (Show, Eq)
+
+instance Exception TestError
+
+prop_verify_connect_to_simulation
+  :: (forall s . IOSim s (Either SomeException ()))
+  -> Property
+prop_verify_connect_to_simulation sim =
+  let tr = runSimTrace sim
+   in case traceResult True tr of
+     Left e -> counterexample
+                (unlines
+                  [ "=== Say Events ==="
+                  , unlines (selectTraceEventsSay' tr)
+                  , "=== Trace Events ==="
+                  , unlines (show `map` traceEvents tr)
+                  , "=== Error ==="
+                  , show e ++ "\n"
+                  ])
+                False
+     Right (Left e) -> counterexample
+                        (unlines
+                          [ "=== Say Events ==="
+                          , unlines (selectTraceEventsSay' tr)
+                          , "=== Trace Events ==="
+                          , unlines (show `map` traceEvents tr)
+                          , "=== Error ==="
+                          , show e ++ "\n"
+                          ])
+                        False
+     Right (Right _) -> property True
+
+prop_connect_to_accepting_socket :: BearerInfoScript -> Property
+prop_connect_to_accepting_socket (BearerInfoScript script) =
+    prop_verify_connect_to_simulation sim
```

Comment:
I think we should verify that the connection was established by looking at the net-sim state, rather than just check that there's no error.

Reply:
I'll rebase this PR on #3417 to get access to `UniverseState`